### PR TITLE
Use LoadCIF in Engineering GSAS-II tab and fix bugs relating to multiple phases

### LINKED ISSUE #37109: Out of index error reading elements from cif file  in GSAS-II tab of EngDiff UI
**Original reporter:** Joe Kelleher (ENGIN-X)

**Describe the bug**
Recorded in support issue
https://github.com/ISISNeutronMuon/mantid-isis-support/issues/190

**To Reproduce**
(1) Open EngDiff GUI
(2) Enter paths to instrument .prm and fpcuissed data
(3) Enter path to this cif file for phase (renamed from .cif)
[tum-2_b.cif.txt](https://github.com/mantidproject/mantid/files/14852675/tum-2_b.cif.txt)
(4) Click refine

```
Traceback (most recent call last):
  File "c:\mantid-conda\mantid\qt\python\mantidqtinterfaces\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\gsas2\presenter.py", line 37, in on_refine_clicked
    number_output_histograms = self.model.run_model(
  File "c:\mantid-conda\mantid\qt\python\mantidqtinterfaces\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\gsas2\model.py", line 105, in run_model
    self.evaluate_further_inputs(user_limits)
  File "c:\mantid-conda\mantid\qt\python\mantidqtinterfaces\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\gsas2\model.py", line 151, in evaluate_further_inputs
    if not self.loop_phase_files() or not self.validate_x_limits(user_x_limits):
  File "c:\mantid-conda\mantid\qt\python\mantidqtinterfaces\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\gsas2\model.py", line 162, in loop_phase_files
    generated_reflections = self.generate_reflections_from_space_group(loop_phase_filepath, self.chosen_cell_lengths)
  File "c:\mantid-conda\mantid\qt\python\mantidqtinterfaces\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\gsas2\model.py", line 481, in generate_reflections_from_space_group
    found_basis = self.read_basis(phase_filepath)
  File "c:\mantid-conda\mantid\qt\python\mantidqtinterfaces\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\gsas2\model.py", line 356, in read_basis
    split_line = self.format_element_symbol(split_line)
  File "c:\mantid-conda\mantid\qt\python\mantidqtinterfaces\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\gsas2\model.py", line 373, in format_element_symbol
    line_split[0] = "".join([i for i in line_split[0] if not i.isdigit()])
IndexError: list index out of range
```

**Expected behavior**
No crash

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
### Description of work

GSAS-II tab in the EngDiff UI had some flaky python code to read .cif files (resulting in bugs like the original issue #37109).
This PR replaces this code with a call to the algorithm `LoadCIF` which is more robust when handling e.g. duplicate keys, multiple ways of defining the spacegroup etc.

In testing this it became apparent there were a couple of other bugs that could be fixed in the same area of code:

- User supplied lattice parameters were applied to all phases (are now only applied to the first phase)
- Existing code was not generating reflection for multiple phases correctly, and indeed was passing all the reflection for all phases to each individual phase in the GSAS-II API.

Fixes #37109

**Report to:** Joe Kelleher (ENGIN-X)

### To test:
(1) Follow instructions for Test 11 in the manual testing docs

https://developer.mantidproject.org/Testing/EngineeringDiffraction/EngineeringDiffractionTestGuide.html#test-11

(2) Override the lattice parameter to be something very different - e.g 4.1 Ang and click Refine (fit should not look good, and the lattice parameter table will have a value similar to the entered value)

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/call_G2sc.py
Comment (warunawickramasingha) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/call_G2sc.py:184]: I have installed latest GSAS-II for windows from https://subversion.xray.aps.anl.gov/trac/pyGSAS/wiki/InstallWindows. I am getting below error when `Refine in GSAS` is clicked. 
Note, in my installation `gsas2full\GSAS-II\GSASII` path does not exist and it is `gsas2full\GSASII` directory to find `GSASIIscriptable.py`

```
GSAS-II call failed with error: Traceback (most recent call last):
  File "C:\Codes\mantid_for_Testing\qt\python\mantidqtinterfaces\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\gsas2\call_G2sc.py", line 186, in <module>
    import GSASIIscriptable as G2sc
ModuleNotFoundError: No module named 'GSASIIscriptable'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Codes\mantid_for_Testing\qt\python\mantidqtinterfaces\mantidqtinterfaces\Engineering\gui\engineering_diffraction\tabs\gsas2\call_G2sc.py", line 188, in <module>
    raise ImportError(f"GSAS-II was not found at {import_path}")
ImportError: GSAS-II was not found at C:/Users/kli94267/AppData/Local/gsas2full\GSAS-II\GSASII
```

Comment (RichardWaiteSTFC) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/call_G2sc.py:184]: Yes sorry, I had to re-install GSAS and I think in later versions the path has changed - but I didn't mean to include this change in the PR as it is currently working on IDAaaS. I'll change it back, but for now if you could correct this path and continue with the testing that'd be great thanks!

**Review by warunawickramasingha — APPROVED** [review]
After changing the GSASIIscriptable path locally, all seems to be working fine as expected. 
Will approve after the above path is corrected.

**Comment by RichardWaiteSTFC** [comment]
Woops, sorry I didn't get round to reverting GSAS path before this was merged - I'll open a quick separate PR to revert it shortly!

**Pull request merged**
